dev/tasks/macros.jinja

```diff
@@ -137,11 +137,25 @@ on:
     run: |
       sudo apt update
       sudo apt install -y ruby-full
+  - name: Set up Ruby by GitHub Actions
+    if: runner.arch == 'X64' && runner.os != 'macOS'
+    uses: ruby/setup-ruby@v1
+    with:
+      ruby-version: "ruby"
+  - name: Install gemfury client on ARM self-hosted
+    if: runner.arch != 'X64'
+    run: |
+      # GH-36692: Pin gemfury due to wrong faraday dependency declaration.
+      gem install --user-install gemfury -v 0.12.0
```

Comment:
Is there a downside to  the user install option? If not should we just install that way on all runners?
Alternatively these steps could be merged and the option inserted like this `{{ "${{ runner.arch == 'X64' && '--user-install' || ''}}" }}`

Reply:
If we want to use the `setup-ruby` action, to ensure recent ruby versions, the step fails when trying to use `--user-install` that's why we are separating them. As suggested by Kou I've also added adding the user install dir to the path on the ARM step.
I also prefer to be more explicit on the difference between ARM and non-ARM on the different step. I feel is more readable than:
`{{ "${{ runner.arch == 'X64' && '--user-install' || ''}}" }}`